src/hosted-buttons/utils.js

```diff
@@ -220,3 +225,86 @@ export const buildHostedButtonOnApprove = ({
     });
   };
 };
+
+export function getFlexDirection({
+  layout,
+}: GetFlexDirectionArgs): GetFlexDirection {
+  return { flexDirection: layout === "horizontal" ? "row" : "column" };
+}
+
+export function shouldRenderSDKButtons(
+  fundingSources: $ReadOnlyArray<FundingSources>
+): boolean {
+  return Boolean(fundingSources?.length);
+}
+
+export function getButtonColor(
+  color: Color,
+  fundingSource: FundingSources
+): Color {
+  const colorMap = {
+    gold: {
+      paypal: "gold",
+      venmo: "blue",
+      paylater: "gold",
+    },
+    blue: {
+      paypal: "blue",
+      venmo: "silver",
+      paylater: "blue",
+    },
+    black: {
+      paypal: "black",
+      venmo: "black",
+      paylater: "black",
+    },
+    white: {
+      paypal: "white",
+      venmo: "white",
+      paylater: "white",
+    },
+    silver: {
+      paypal: "silver",
+      venmo: "blue",
+      paylater: "silver",
+    },
+  };
+
+  return colorMap[color][fundingSource];
+}
+
+export function buildButtonContainer({
+  flexDirection,
+  selector,
+}: BuildButtonContainerArgs) {
+  if (typeof selector !== "string") {
+    throw new TypeError("Selector must be a string");
+  }
+
+  if (selector.charAt(0) !== "#") {
+    throw new TypeError("Selector must be referring to an id");
+  }
```

Comment:
we should consider doing something like this index.js:
```
const parsedSelector = parseSelector(selector)
...
func parsedSelector (selector: ?string) {
  ...validation checks on selector
}
```

That way we return very early if we weren't given a proper selector and any code that gets the `parsedSelector` knows its good

Reply:
I refactored this function to accept html elements. I've created a new check to see if the element was found and throw and error if not.

Let me know if you think it still should be its own function

https://github.com/paypal/paypal-checkout-components/pull/2375/files#diff-dc0ab5efddb0f4c5dc8c699ccafce1daa87906e1b839480c5be57048cee5482bR285